### PR TITLE
Graphical file picker using Zenity

### PR DESCRIPTION
**Pull request by kmafeni04**
With help from @Guldoman, made a plugin that allows for a graphical file picker using the zenity program on linux

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1471,6 +1471,13 @@
       "name": "wal.lxl",
       "mod_version": "3",
       "replaces": [ "wal_lxl" ]
-    }
+    },
+        {
```
Comment (Guldoman) [inline manifest.json:1475]: Would you mind aligning this?

File: plugins/zenity.lua [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+			doc:save(abs_path, abs_path)
+			core.log("Saved as " .. '"' .. dv.doc.filename  .. '"')
+		end)
+	end,
+
+	["zenity:save"] = function(dv)
+		local doc = dv.doc
+		if dv.doc.filename then
+			doc:save()
+			core.log("Saved " .. '"' .. dv.doc.filename  .. '"')
```
Comment (Guldoman) [inline plugins/zenity.lua:199]: I guess you could do
```suggestion
			command.perform("doc:save")
```

File: plugins/zenity.lua [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+			end
+			local abs_path = table.concat(buffer) or ""
+
+			-- Remove final newline zenity adds
+			abs_path = string.match(abs_path, "^[^\n]+") or ""
+			if #abs_path == 0 then
+				return
+			end
+			doc:save(abs_path, abs_path)
+			core.log("Saved as " .. '"' .. dv.doc.filename  .. '"')
```
Comment (Guldoman) [inline plugins/zenity.lua:191]:
```suggestion
			core.log('Saved as "%s"', dv.doc.filename)
```

File: manifest.json [hunk 4]
```diff
@@ -1471,6 +1471,13 @@
...
+      "description": "Graphical filepicker using zenity.",
+      "version": "0.1",
+      "path": "plugins/zenity.lua",
+      "id": "zenity",
+      "mod_version": "3"
+    },
```
Comment (Guldoman) [inline manifest.json:1481]:
```suggestion
    }
```
Trailing `,` breaks JSON.

**Review by Guldoman — CHANGES_REQUESTED** [review]
The trailing comma in the manifest is an issue, as it'll break `lpm`.

There is a bunch of repeated code that could be moved into a function, but it's alright for now.

I'm not sold on the plugin ID/filename/command prefixes being just `zenity` tho.

**Comment by kmafeni04** [comment]
> I'm not sold on the plugin ID/filename/command prefixes being just `zenity` tho.

What could be a more suitable name, `gui_save`?

Also will make changes in the morning

**Comment by Guldoman** [comment]
> What could be a more suitable name, `gui_filepicker`?

Yeah, that sounds a bit more descriptive.
I wonder if we should note that `zenity` is only available on linux (and kinda on macOS too).

**Comment by kmafeni04** [comment]
> Yeah, that sounds a bit more descriptive. I wonder if we should note that `zenity` is only available on linux (and kinda on macOS too).

That should be right

**Pull request merged**